### PR TITLE
Post Visibility e2e test: Improve XPath selector to avoid reliance on DOM structure

### PR DESCRIPTION
**Pull request by chad1008**
## Description
As discussed in https://github.com/WordPress/gutenberg/pull/37491#issuecomment-1034913604, this PR modifies the XPath selector targeting a future date for the post's publication when running post visibility e2e tests. The new XPath makes the test more resilient by avoiding reliance on DOM structure which could change in the future.

## Testing Instructions
From your local Gutenberg directory, run
`npx wp-scripts test-e2e --config packages/e2e-tests/jest.config.js packages/e2e-tests/specs/editor/various/post-visibility.test.js`. All tests should pass, specifically final test related to changing post visibility when the publication date is in the future

cc @ciampo 

## Types of changes
enhancement

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/post-visibility.test.js [hunk 1]
```diff
@@ -95,9 +95,7 @@ describe( 'Post visibility', () => {
 			'div[aria-label="Move forward to switch to the next month."]'
 		);
 		await (
-			await page.$x(
-				'//td[contains(concat(" ", @class, " "), " CalendarDay ")]/div[contains(concat(" ", @class, " "), " components-datetime__date__day ")][text() = "15"]'
-			)
+			await page.$x( '//td[@role="button"]/*[text() = "15"]' )
```
Comment (ciampo) [inline packages/e2e-tests/specs/editor/various/post-visibility.test.js]: Maybe we could make this a little bit more robust by adding, at the start of the selector, a parent element with `role="application"` and `aria-label="Calendar"` ?

Screenshot 2022-02-17 at 22 11 40


We could probably apply this improvement also to the e2e tests added in #37491

Comment (chad1008) [reply packages/e2e-tests/specs/editor/various/post-visibility.test.js]: Ah, well spotted! It's done, and I'll put a separate PR together for the other tests you mentioned 😄

**Review by ciampo — APPROVED** [review]
LGTM 🚀  

This PR may need a rebase to get rid of the failing e2e test

**Comment by chad1008** [comment]
rebased and passing! ✅

**Comment by ciampo** [comment]
The CI failure (the Gradle cache can't be restored when running React Native E2E tests)  doesn't seem related to the changes in this PR, I'm going to merge

**Pull request merged**